File: src/main/java/com/sarthak/agenticai/service/WishlistService.java
package com.sarthak.agenticai.service;

import com.sarthak.agenticai.dto.WishlistResponseDto;

import java.util.List;

public interface WishlistService {

    WishlistResponseDto addToWishlist(
            String email,
            Long productId);

    List<WishlistResponseDto> getWishlist(
            String email);

    void removeFromWishlist(
            String email,
            Long wishlistItemId);

    void clearWishlist(
            String email);
}

File: agentic-ai-frontend/src/services/wishlistService.js
// src/services/wishlistService.js

import api from "../api/axios";

// ========================================
// ADD TO WISHLIST
// ========================================

export const addToWishlist = async (
    email,
    productId
) => {

    const response = await api.post(
        "/wishlist",
        {
            productId
        },
        {
            params: {
                email
            }
        }
    );

    return response.data;
};


// ========================================
// GET WISHLIST
// ========================================

export const getWishlist = async (email) => {

    const response = await api.get(
        "/wishlist",
        {
            params: {
                email
            }
        }
    );

    return response.data;
};


// ========================================
// REMOVE FROM WISHLIST
// ========================================

export const removeFromWishlist = async (
    email,
    wishlistItemId
) => {

    await api.delete(
        `/wishlist/${wishlistItemId}`,
        {
            params: {
                email
            }
        }
    );
};


// ========================================
// CLEAR WISHLIST
// ========================================

export const clearWishlist = async (email) => {

    await api.delete(
        "/wishlist/clear",
        {
            params: {
                email
            }
        }
    );
};